autokeras/hypermodel/hyper_node.py

```diff
@@ -26,20 +27,34 @@ def __init__(self, **kwargs):
     def build(self, hp):
         return tf.keras.Input(shape=self.shape)
 
+    @staticmethod
+    def related_block():
+        return hyper_block.GeneralBlock()
 
-class ImageInput(Node):
 
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
+class ImageInput(Input):
 
-    def build(self, hp):
-        return tf.keras.Input(shape=self.shape)
+    @staticmethod
+    def related_block():
```

Comment:
Maybe you want to make `related_block` an abstract static method under `Input`, then extract out an `GeneralInput` inherits under `Input`